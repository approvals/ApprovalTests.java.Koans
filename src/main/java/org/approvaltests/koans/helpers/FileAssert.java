package org.approvaltests.koans.helpers;

import org.junit.Assert;

import com.spun.util.ObjectUtils;
import com.spun.util.io.FileUtils;

public class FileAssert
{
  public static void verifyContentsIsEqual(String fileContainingExpected, String actual)
  {
    try
    {
      StackTraceElement caller = Thread.currentThread().getStackTrace()[2];
      Class<?> clazz = Class.forName(caller.getClassName());
      String expected = FileUtils.readFromClassPath(clazz, fileContainingExpected).trim();
      Assert.assertEquals(expected, actual);
    }
    catch (Exception e)
    {
      throw ObjectUtils.throwAsError(e);
    }
  }
}
