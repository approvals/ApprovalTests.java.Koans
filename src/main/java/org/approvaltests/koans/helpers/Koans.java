package org.approvaltests.koans.helpers;

import java.io.File;

import com.spun.util.ClassUtils;

public class Koans
{
  public String ___  = "Please fill in the blank";
  public int    ____ = 0;
  
  public String decode(String string)
  {
    int offshift = 9;
    return push(string, offshift);
  }
  
  public String push(String string, int offshift)
  {
    StringBuffer b = new StringBuffer();
    for (char c : string.toCharArray())
    {
      b.append((char) (c + offshift));
    }
    return b.toString();
  }
  
  public String getPath(String file)
  {
    return ClassUtils.getSourceDirectory(getClass()).getAbsolutePath() + File.separator + file;
  }
}
