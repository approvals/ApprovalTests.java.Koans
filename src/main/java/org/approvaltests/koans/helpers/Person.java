package org.approvaltests.koans.helpers;

public class Person
{
  private final String  firstName;
  private final String  lastName;
  private final boolean isMale;
  private final int     age;

  public Person(String firstName, String lastName, boolean isMale, int age)
  {
    this.firstName = firstName;
    this.lastName = lastName;
    this.isMale = isMale;
    this.age = age;
  }

  public String getFirstName()
  {
    return firstName;
  }

  public String getLastName()
  {
    return lastName;
  }

  public boolean isMale()
  {
    return isMale;
  }

  public int getAge()
  {
    return age;
  }
}
