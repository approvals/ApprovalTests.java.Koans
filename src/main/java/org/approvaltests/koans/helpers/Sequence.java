package org.approvaltests.koans.helpers;

import java.text.MessageFormat;

import org.lambda.functions.Function1;

public class Sequence implements Function1<Integer, String>
{
  @Override
  public String call(Integer i)
  {
    int n1 = 1;
    int n2 = 1;
    for (int j = 2; j <= i; j++)
    {
      int n3 = n1 + n2;
      n1 = n2;
      n2 = n3;
    }
    return MessageFormat.format("#{0} => {1}", i, n2);
  }
}
