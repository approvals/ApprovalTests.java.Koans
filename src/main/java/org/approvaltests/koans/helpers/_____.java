package org.approvaltests.koans.helpers;

import org.junit.Assert;

import org.approvaltests.core.ApprovalFailureReporter;

public class _____ implements ApprovalFailureReporter
{
  @Override
  public void report(String received, String approved) throws Exception
  {
    Assert.fail("You need to fill in the blank");
  }
}
