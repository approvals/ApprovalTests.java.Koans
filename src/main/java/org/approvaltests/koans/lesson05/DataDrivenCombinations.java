package org.approvaltests.koans.lesson05;

import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.koans.helpers.Koans;
import org.junit.Test;

import java.text.SimpleDateFormat;
import java.time.LocalDateTime;
import java.time.Month;
import java.time.ZoneId;
import java.time.format.DateTimeFormatter;
import java.util.Date;

/**
 * How to do Koans:
 * Step 1: Press the Run Button (Place cursor on the Method name to run a single method)
 * PC: Ctrl+F11
 * Mac: Command+fn+F11
 * Step 2: Read the name of the Method that Failed
 * Step 3: Fill in the blank (___) to make it pass
 * Step 4: Repeat Until Enlightenment
 * Do not change anything except the blank (___)
 */

public class DataDrivenCombinations extends Koans {

    @Test
    public void oneDimensionalInputIsTheSameAsArray() throws Exception {
        String[] beatles = {"John", ___, "Paul", "George", "Ringo"};
        CombinationApprovals.verifyAllCombinations(a -> a, beatles);
    }

    @Test
    public void twoDimensionalInputGoesByTheSquare() throws Exception {
        Integer[] small = {1, 2, 3};
        Integer[] big = {11, ____, 55};

        CombinationApprovals.verifyAllCombinations((a, b) -> a * b, small, big);
    }

    @Test
    public void exceptionsAreOk() throws Exception {
        Integer[] numerator = {60, 126, 42};
        Integer[] denominator1 = {3, 2};
        Integer[] denominator2 = {-1, ____, 1};

        CombinationApprovals.verifyAllCombinations((a, b, c) -> a / (b + c), numerator, denominator1, denominator2);
    }

    @Test
    public void workingWithALambda() throws Exception {
        LocalDateTime newMillennium = LocalDateTime.of(2000, Month.JANUARY, 1, 0, 0, 0);
        String[] years = {"yy", "yyyy"};
        String[] months = {"M", ___, "MMMM"};
        String[] days = {"dd", "EEEE"};

        CombinationApprovals.verifyAllCombinations((y, m, d) ->
                newMillennium.format(DateTimeFormatter.ofPattern(y + "/" + m + "/" + d)), years, months, days);
    }
}

