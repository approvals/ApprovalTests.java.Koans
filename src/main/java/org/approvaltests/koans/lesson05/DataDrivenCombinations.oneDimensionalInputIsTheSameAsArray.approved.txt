[John] => John 
[Yoko] => Yoko 
[Paul] => Paul 
[George] => George 
[Ringo] => Ringo 
